### DISCUSSION TITLE
Support for Google Gemini

### DISCUSSION THREAD

**Discussion by machinewrapped** [machinewrapped/llm-subtrans#131] [General]
This release adds Google Gemini as a translation provider. 

**Please note that the Gemini API can only be accessed from IP addresses in certain geographic regions at the moment:** https://ai.google.dev/available_regions

To use Gemini as a translator you will need an API key. I believe there is a simplified procedure for getting one at https://ai.google.dev/ if you are in a supported region. Alternatively you can set up a project on https://console.cloud.google.com/ and generate an API key. You must ensure that Generative AI is enabled for the project and the key.

Google have only exposed the Gemini 1.0 Pro model via the API at the moment, though you can sign on to a waiting list to access to the new 1.5 Pro. In my tests Gemini 1.0 Pro is comparable to or perhaps a little better than the latest GPT 3.5 model from OpenAI when it comes to subtitle translation, and is basically free to use for this purpose (https://ai.google.dev/pricing).

This will be a pre-release/beta version to encourage testing and exploration. Please drop a note in discussions if you are able to use Gemini and let us know how your experience was, or create an issue if you encounter any problems (other than Gemini not being available in your region).

<hr /><em>This discussion was created from the release <a href='https://github.com/machinewrapped/gpt-subtrans/releases/tag/v0.6.1'>Support for Google Gemini</a>.</em>

**Comment by hnorgaar** [comment]
Sometimes I get the batch description in the subtitles

**Comment by machinewrapped** [comment]
Yeah, the batch or scene description appearing in the subtitles seems to be an issue with Gemini. I think it happens when the response gets truncated due to length, leaving an unclosed tag. I need to add some extra validation around that.

I've fixed the issue with provider temperature (and rate limit) getting reset to their default values in v0.6.2:
https://github.com/machinewrapped/gpt-subtrans/releases/tag/v0.6.2

**Reply by hnorgaar** [reply]
Thx, but the only thing the GUI load/shows is temperature for OpenAI, the other sets back to 0.00 in the GUI in 6.2

**Reply by machinewrapped** [reply]
Confirmed, will be fixed in the next release

**Reply by machinewrapped** [reply]
Fixed in v0.6.3 (I hope): https://github.com/machinewrapped/gpt-subtrans/releases

**Comment by machinewrapped** [comment]
I've found that Gemini is better than GPT at making corrections when provided with source subtitles that contain errors, e.g. those generated with Whisper. It seems to be quite good at figuring out what the correct word should be when a lookalike or soundalike word has been used.

If you have subtitles which are likely to contain errors try using instructions (whispered) or instructions (OCR errors) instead of the default instructions - they're almost identical (to each other), just including a hint as to the likely source of errors.

**Reply by hnorgaar** [reply]
Thx for the response and keep up the good work. Its a very usefull tool for sub translation

**Comment by machinewrapped** [comment]
Gemini 1.5 Pro is now available via the API.
